cwd/editor.js

```diff
@@ -0,0 +1,223 @@
+    "type": "POST",
+    "timeout": 0,
+    "headers": {
+      "Content-Type": "application/x-www-form-urlencoded"
+    },
+    "data": {
+      "message": "\"" + $('#input-' + (btn.id.substring(8, ))).val() + "\"",
+      "probability": `\[${prob}\]`,
+      "pass": "\"" + $('#passIn').val() + "\""
+    }
+  };
+
+  $.ajax(settings).done(function(response) {
+    console.log(response);
+  });
+}
+
+const updateView = (btn) => {
+  const viewNum = btn.id.substring(8, 9);
+  const msgNum = btn.id.substring(10, 11);
+
+  const prob = [];
+  for (const x of Array(1).keys()) {
+    prob.push($('#prob-' + (btn.id.substring(8, )) + "-" + (x + 1)).val());
+  }
+
+  const settings = {
+    "url": views[viewNum - 1] + "gauges/0/messages/" + msgNum,
+    "type": "POST",
+    "timeout": 0,
+    "headers": {
+      "Content-Type": "application/x-www-form-urlencoded"
+    },
+    "data": {
+      "message": "\"" + $('#input-' + (btn.id.substring(8, ))).val() + "\"",
+      "probability": `\[${prob}\]`,
+      "pass": "\"" + $('#passIn').val() + "\""
+    }
+  };
+
+  $.ajax(settings).done(function(response) {
+    console.log(response);
+  });
+}
+
+const makeButton = (id, div, text, ref) => {
+  let upBtn = document.createElement("BUTTON");
+  upBtn.setAttribute("type", "button");
+  upBtn.setAttribute("id", id)
+  upBtn.innerHTML = text
+
+  if (ref) {
+    div.insertBefore(upBtn, ref)
+  } else {
+    div.appendChild(upBtn);
+  }
+};
+
+const makeRadioButton = (id, div, text) => {
+  let radBtn = document.createElement("INPUT");
+  radBtn.setAttribute("type", "radio");
+  radBtn.setAttribute("id", id);
+  radBtn.setAttribute("name", "upload");
+
+  let label = document.createElement("LABEL");
+  label.setAttribute("for", id);
+  label.innerHTML = text;
+
+  div.appendChild(label);
+  div.appendChild(radBtn);
+};
+
+const makeTextInput = (id, div, text, ref) => {
+  let messageText = document.createElement("INPUT");
+  messageText.setAttribute("id", id);
+  messageText.setAttribute("type", "text");
+  messageText.setAttribute("value", text);
+  messageText.setAttribute("size", 85);
+
+  if (ref) {
+    div.insertBefore(messageText, ref)
+  } else {
+    div.appendChild(messageText);
+  }
+};
+
+const makeNumInput = (id, div, num, ref) => {
+  let messageProb = document.createElement("INPUT");
+  messageProb.setAttribute("id", id);
+  messageProb.setAttribute("type", "number");
+  messageProb.setAttribute("value", num);
+  messageProb.setAttribute("size", 1);
+
+  if (ref) {
+    div.insertBefore(messageProb, ref)
+  } else {
+    div.appendChild(messageProb);
+  }
+}
+
+let viewItr = 0;
+const views = ['http://localhost:3006/glyphs/5d3744e584687c30ef9e5b47/', 'http://localhost:3006/glyphs/5d37452784687c30ef9e5b48/',
+  'http://localhost:3006/glyphs/5d4885f18227675bc73b98ac/'
+]
+
+views.forEach(view => {
+  $.getJSON(view, function(data) {
+    let viewMsgItr = 0;
+    viewItr++;
```

Comment:
Why not just have it start at 1 then?

Reply:
Have it start at 1 and put the iterator at the end or start at 0 and have it like it is. What's the functional difference?

Reply:
I guess I am just confused by the scope of `viewMsgItr`.  If this is supposed to be the index of each `view`, why not just have `forEach` provide an index itself?

Reply:
The scope is still confusing, but this doesn't matter if it works fine.